File: HttpDnsDemo_Android/HttpDnsDemo/src/com/tencent/msdk/dns/MainActivity.java
package com.tencent.msdk.dns;

import java.util.Vector;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.tencent.beacon.event.UserAction;

public class MainActivity extends Activity {
	private EditText mEdit;
	private Button mButton;
	private TextView mText;
	private String httpDns;
	private Vector<String> vector = null;
	private static final int UPDATE_UI = 0;
	private static String TAG = "WGGetHostByName";
	private String mDomain = "www.qq.com.";

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		mEdit = (EditText) findViewById(R.id.edit);
		mButton = (Button) findViewById(R.id.button);
		mText = (TextView) findViewById(R.id.text);

		// 初始化HttpDns
		MSDKDnsResolver.getInstance().init(MainActivity.this.getApplicationContext());
		// 初始化灯塔（注意：如果已接入MSDK或者灯塔，检查是否已初始化，以免重复操作）
		UserAction.initUserAction(MainActivity.this.getApplicationContext());

		mButton.setOnClickListener(new View.OnClickListener() {
			public void onClick(View v) {
				new Thread(new Runnable() {
					@Override
					public void run() {
						// mDomain只能传入域名请不要传入ip
						mDomain = mEdit.getText().toString();
						Log.i(TAG, "mDomain is " + mDomain);
						if (mDomain != null) {
							vector = WGGetHostByName(mDomain);
							if (vector != null) {
								httpDns = vector.toString().replace("[", "").replace("]", "");
								Log.i(TAG, "httpDns is " + httpDns);

							}
						}

						Message msg = new Message();
						msg.what = UPDATE_UI;
						msg.obj = httpDns;
						mHandler.sendMessage(msg);

					}
				}).start();

			}
		});

	}

	@SuppressWarnings({ "unchecked", "rawtypes" })
	public static Vector<String> WGGetHostByName(String domain) {
		// 调用httpdns解析接口
		String ips = MSDKDnsResolver.getInstance().getAddrByName(domain);
		Vector mVector = new Vector();
		if (ips != null) {
			if (ips.contains(";")) {
				String[] temp = ips.split(";");
				for (int i = 0; i < temp.length; i++) {
					mVector.add(temp[i]);
				}
			} else {
				mVector.add(ips);
			}
		}
		return mVector;
	}

	@SuppressLint("HandlerLeak")
	Handler mHandler = new Handler() {
		public void handleMessage(Message msg) {
			super.handleMessage(msg);
			switch (msg.what) {
			case UPDATE_UI:
				String data = (String) msg.obj;
				mText.setText(data);
				break;
			default:
				break;
			}
		}

	};

}

File: HttpDnsSample/app/src/main/java/com/yx/dnsTest/MainActivity.kt
package com.yx.dnsTest

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.tencent.msdk.dns.DnsConfig
import com.tencent.msdk.dns.MSDKDnsResolver

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val TAG = "MainActivity"

        val textInput: EditText = findViewById<EditText>(R.id.textInput)
        val btn = findViewById<Button>(R.id.button)
        val btnAsync = findViewById<Button>(R.id.buttonAsync)
        val btnBatch = findViewById<Button>(R.id.buttonBatch)
        val btnAsyncBatch = findViewById<Button>(R.id.buttonAsyncBatch)
        val resultText = findViewById<TextView>(R.id.resultText)

        // 配置项
        val dnsConfigBuilder = DnsConfig.Builder()
            .dnsId("dnsId")
            .dnsIp("119.29.29.98")
            .desHttp()
            .dnsKey("dnsKey")
            .logLevel(Log.VERBOSE)
            .timeoutMills(1000)
            .preLookupDomains("xxx.com")
            .enableReport(true)
            .build()
        MSDKDnsResolver.getInstance().init(this, dnsConfigBuilder)

        //  域名解析
        btn.setOnClickListener {
            val hostname = textInput.getText().toString();

            // 进⾏行行域名解析
            // NOTE: ***** 域名解析接⼝口是耗时同步接⼝口，不不应该在主线程调⽤用 *****
            // useHttp即是否通过HTTP协议访问HTTP服务进⾏行行域名解析
            // useHttp为true时通过HTTP协议访问HTTP服务进⾏行行域名解析, 否则通过UDP协议访问HTTP服务 进⾏行行域名解析
            // ipSet即解析得到的IP集合
            // ipSet.v4Ips为解析得到IPv4集合, 可能为null
            // ipSet.v6Ips为解析得到IPv6集合, 可能为null
            Log.d(TAG, "HTTPDNS：" + hostname)

            Thread(Runnable {
                val start_time = System.currentTimeMillis();
                val ips = MSDKDnsResolver.getInstance().getAddrByName(hostname);
                runOnUiThread {
                    Log.d(TAG, "HTTPDNS：" + hostname + ": " + ips)
                    resultText.setText("解析结果为：" + ips + ", elapse:" + (System.currentTimeMillis() - start_time));
                }
            }).start()
        }

        //  域名解析-批量
        btnBatch.setOnClickListener {
            val hostname = textInput.getText().toString();

            // 进⾏行行域名解析
            // NOTE: ***** 域名解析接⼝口是耗时同步接⼝口，不不应该在主线程调⽤用 *****
            // useHttp即是否通过HTTP协议访问HTTP服务进⾏行行域名解析
            // useHttp为true时通过HTTP协议访问HTTP服务进⾏行行域名解析, 否则通过UDP协议访问HTTP服务 进⾏行行域名解析
            // ipSet即解析得到的IP集合
            // ipSet.v4Ips为解析得到IPv4集合, 可能为null
            // ipSet.v6Ips为解析得到IPv6集合, 可能为null
            Log.d(TAG, "HTTPDNS：" + hostname)

            Thread(Runnable {
                val start_time = System.currentTimeMillis();
                val ips = MSDKDnsResolver.getInstance().getAddrsByName(hostname);
                val end_time = System.currentTimeMillis() - start_time;
                runOnUiThread {
                    Log.d(TAG, "HTTPDNS：" + hostname + ": " + ips)
                    resultText.setText("解析结果为：" + ips + ", 耗时:" + end_time);
                }
            }).start()
        }

        //  异步回调
        MSDKDnsResolver.getInstance()
            .setHttpDnsResponseObserver { tag, domain, ipResultSemicolonSep ->
                Log.d(TAG, "tag:" + tag);
                val elapse = System.currentTimeMillis() - tag.toLong();
                val lookedUpResult =
                    "[[getAddrByNameAsync]]:ASYNC:::" + ipResultSemicolonSep +
                            ", domain:" + domain + ", tag:" + tag +
                            ", elapse:" + elapse
                Log.d(TAG, "HTTPDNS_Async：" + lookedUpResult);
                resultText.setText("解析结果为：" + ipResultSemicolonSep + ", elapse:" + elapse);
            }

        //  异步解析
        btnAsync.setOnClickListener(View.OnClickListener {
            val hostname = textInput.getText().toString();

            MSDKDnsResolver.getInstance()
                .getAddrByNameAsync(hostname, System.currentTimeMillis().toString())
        })

        // 异步解析-批量
        btnAsyncBatch.setOnClickListener(View.OnClickListener {
            val hostname = textInput.getText().toString();

            MSDKDnsResolver.getInstance()
                .getAddrsByNameAsync(hostname, System.currentTimeMillis().toString())
        })

    }
}